src/pages/workspace/WorkspaceSettingsCurrencyPage.js

```diff
@@ -0,0 +1,106 @@
+import ScreenWrapper from '../../components/ScreenWrapper';
+import HeaderWithBackButton from '../../components/HeaderWithBackButton';
+import SelectionList from '../../components/SelectionList';
+import Navigation from '../../libs/Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+import compose from '../../libs/compose';
+import ONYXKEYS from '../../ONYXKEYS';
+import withPolicy, {policyDefaultProps, policyPropTypes} from './withPolicy';
+import * as Policy from '../../libs/actions/Policy';
+
+const propTypes = {
+    /** Constant, list of available currencies */
+    currencyList: PropTypes.objectOf(
+        PropTypes.shape({
+            /** Symbol of the currency */
+            symbol: PropTypes.string,
+        }),
+    ),
+    ...policyPropTypes,
+};
+
+const defaultProps = {
+    currencyList: {},
+    ...policyDefaultProps,
+};
+
+function WorkspaceSettingsCurrencyPage({currencyList, policy}) {
+    const {translate} = useLocalize();
+    const [searchText, setSearchText] = useState('');
+
+    const getDisplayText = useCallback((currencyCode, currencySymbol) => `${currencyCode} - ${currencySymbol}`, []);
+
+    const {sections, initiallyFocusedOptionKey} = useMemo(() => {
+        const trimmedText = searchText.trim().toLowerCase();
+        const currencyListKeys = _.keys(currencyList);
+
+        const filteredItems = _.filter(currencyListKeys, (currencyCode) => {
+            const currency = currencyList[currencyCode];
+            return getDisplayText(currencyCode, currency.symbol).toLowerCase().includes(trimmedText);
+        });
+
+        let selectedCurrencyCode;
+
+        const currencyItems = _.map(filteredItems, (currencyCode) => {
+            const currency = currencyList[currencyCode];
+            const isSelected = policy.outputCurrency === currencyCode;
+
+            if (isSelected) {
+                selectedCurrencyCode = currencyCode;
+            }
+
+            return {
+                text: getDisplayText(currencyCode, currency.symbol),
+                keyForList: currencyCode,
+                isSelected,
+            };
+        });
+
+        return {
+            sections: [{data: currencyItems, indexOffset: 0}],
+            initiallyFocusedOptionKey: selectedCurrencyCode,
+        };
+    }, [getDisplayText, currencyList, policy.outputCurrency, searchText]);
+
+    const headerMessage = Boolean(searchText.trim()) && !sections[0].data.length ? translate('common.noResultsFound') : '';
+
+    const onSelectCurrency = (item) => {
+        Policy.updateGeneralSettings(policy.id, policy.name, item.keyForList);
+        Navigation.goBack(ROUTES.getWorkspaceSettingsRoute(policy.id));
+    };
+
+    return (
+        <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            <HeaderWithBackButton
+                title={translate('workspace.editor.currencyInputLabel')}
+                onBackButtonPress={() => Navigation.goBack(ROUTES.getWorkspaceSettingsRoute(policy.id))}
```

Comment:
No stable reference when using inline lambdas, can be wrapped with useCallback :)

Reply:
and maybe use hook instead of importing an object :)

Reply:
@thiagobrez Navigation